precompiles/ArbAggregator.go

```diff
@@ -20,87 +16,111 @@ type ArbAggregator struct {
-	if err != nil || maybeReimbursableAgg == nil {
-		return common.Address{}, false, err
+	if len(posters) == 0 {
+		return common.Address{}, false, errors.New("no batch posters exist")
 	}
-	return *maybeReimbursableAgg, true, nil
+	return posters[0], true, err
 }
 
-// Sets the caller's preferred aggregator to that provided
-func (con ArbAggregator) SetPreferredAggregator(c ctx, evm mech, prefAgg addr) error {
-	var maybePrefAgg *common.Address
-	if prefAgg != (common.Address{}) {
-		maybePrefAgg = &prefAgg
+// [Deprecated]
+func (con ArbAggregator) GetDefaultAggregator(c ctx, evm mech) (addr, error) {
+	posters, err := c.State.L1PricingState().BatchPosterTable().AllPosters()
+	if err != nil {
+		return common.Address{}, err
+	}
+	if len(posters) == 0 {
+		return common.Address{}, errors.New("no batch posters exist")
 	}
-	return c.State.L1PricingState().SetUserSpecifiedAggregator(c.caller, maybePrefAgg)
+	return posters[0], err
 }
 
-// Gets the chain's default aggregator
-func (con ArbAggregator) GetDefaultAggregator(c ctx, evm mech) (addr, error) {
-	return c.State.L1PricingState().DefaultAggregator()
+// Get the addresses of all current batch posters
+func (con ArbAggregator) GetBatchPosters(c ctx, evm mech) ([]addr, error) {
+	return c.State.L1PricingState().BatchPosterTable().AllPosters()
 }
 
-// Sets the chain's default aggregator (caller must be the current default aggregator, its fee collector, or an owner)
+// [Deprecated]
 func (con ArbAggregator) SetDefaultAggregator(c ctx, evm mech, newDefault addr) error {
-	l1State := c.State.L1PricingState()
-	defaultAgg, err := l1State.DefaultAggregator()
+	isOwner, err := c.State.ChainOwners().IsMember(c.caller)
 	if err != nil {
 		return err
 	}
-	allowed, err := accountIsAggregatorOrCollectorOrOwner(c.caller, defaultAgg, c.State)
+	if !isOwner {
+		return ErrNotOwner
+	}
+	batchPosterTable := c.State.L1PricingState().BatchPosterTable()
+	isBatchPoster, err := batchPosterTable.ContainsPoster(newDefault)
 	if err != nil {
 		return err
 	}
-	if !allowed {
-		return errors.New("Only the current default (or its fee collector / chain owner) may change the default")
+	if !isBatchPoster {
+		_, err = batchPosterTable.AddPoster(newDefault, newDefault)
+		if err != nil {
+			return err
+		}
 	}
-	return l1State.SetDefaultAggregator(newDefault)
-}
-
-// Get the aggregator's compression ratio, measured in basis points
-func (con ArbAggregator) GetCompressionRatio(c ctx, evm mech, aggregator addr) (uint64, error) {
-	ratio, err := c.State.L1PricingState().AggregatorCompressionRatio(aggregator)
-	return uint64(ratio), err
+	return nil
 }
 
-// Set the aggregator's compression ratio, measured in basis points
-func (con ArbAggregator) SetCompressionRatio(c ctx, evm mech, aggregator addr, newRatio uint64) error {
-	allowed, err := accountIsAggregatorOrCollectorOrOwner(c.caller, aggregator, c.State)
+func (con ArbAggregator) AddBatchPoster(c ctx, evm mech, newBatchPoster addr) error {
```

Comment:
This method shouldn't be necessary now that batch posters are automatically added.

Reply:
This does have a purpose, for the case where a batch poster wants to set its fee collector address before it posts any batches.  The owner could call this method to create a poster, and then the poster could set its fee collector by calling `SetFeeCollector`.  (We don't allow auto-creation of posters by `SetFeeCollector` because that would allow untrusted parties to create posters.)